src/odemis/driver/test/xt_client_test.py

```diff
@@ -281,6 +283,158 @@ def test_set_beam_shift(self):
+            self.scanner.applyAutoContrastBrightness(detector)
+            starting_time = time.time()
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > 30:
+                    break
+            # Check if the time to perform the autofocus is not to long
+            self.assertLess(time.time() - starting_time, 30, "Execution auto-contrast brightness is slow.")
+            # Line to inspect the execution time to update the expected time
+            print("Execution time was %s seconds " % (time.time() - starting_time))
+            auto_contrast_brightness_state = self.microscope.is_running_auto_contrast_brightness(detector2ChannelName[detector])
+            self.assertEqual(auto_contrast_brightness_state, False)
+
+        # Test if an error is raised if an invalid detector is provided
+        with self.assertRaises(KeyError):
+            self.scanner.applyAutoContrastBrightness("error_expected")
+        time.sleep(2.5)  # Give microscope/simulator the time to update the state
+        autofocus_state = self.microscope.is_autofocusing(detector2ChannelName[detector])
+        self.assertEqual(autofocus_state, False)
+
+    def test_apply_autofocus(self):
+        """
+        Test for the auto functionality of the autofocus.
+        Only checks the returned state, the return object and if an error is raised for an incorrect input state.
+        """
+        # Check all the different detector types by looping over them
+        for detector in detector2ChannelName:
+            autofocus_return = self.efocus.applyAutofocus(detector)
+            time.sleep(2.5)  # Give microscope/simulator the time to update the state
+            autofocus_state = self.microscope.is_autofocusing(detector2ChannelName[detector])
+            self.assertEqual(autofocus_state, True)
+            self.assertIsInstance(autofocus_return, ProgressiveFuture)
+
+            autofocus_return.cancel()
+            time.sleep(5.0)  # Give microscope/simulator the time to update the state
+            autofocus_state = self.microscope.is_autofocusing(detector2ChannelName[detector])
+            self.assertEqual(autofocus_state, False)
+            self.assertIsInstance(autofocus_return, ProgressiveFuture)
+
+            autofocus_return = self.efocus.applyAutofocus(detector)
+            starting_time = time.time()
+            time.sleep(0.5)  # Give microscope/simulator the time to update the state
+            while self.microscope.is_autofocusing(detector2ChannelName[detector]):
+                time.sleep(0.2)  # Wait until the autofocus is finished
+                if time.time() - starting_time > 40:
+                    break
+            # Check if the time to perform the autofocus is not to long
```

Comment:
to -> too

Reply:
See my comment on auto BC regarding definition of "too long" and if you abort the procedure.